bokehjs/src/lib/core/util/bitset.ts

```diff
@@ -94,8 +94,10 @@ export class BitSet implements Equatable {
     return ((this._array[i] >> j) & 0b1) == 0b1
   }
 
-  set(k: number, v: boolean = true): void {
-    this._check_bounds(k)
+  set(k: number, v: boolean = true, c: boolean = true): void {
+    if (c) {
+      this._check_bounds(k)
+    }
```

Comment:
IMO a separate `set_without_bounds_check` would be preferable here. Otherwise using the last parameter means not being able to rely on the default for `v` which is pretty annoying. Also, it's just more explicit and self-explanatory.

Reply:
This definitely should be a separate method and then `set()` should be implemented using it.